Source/Embeddings.Processing/EmbeddingLoopDetector.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Dolittle.Runtime.Projections.Store.State;
+using Dolittle.Runtime.Rudimentary;
+
+namespace Dolittle.Runtime.Embeddings.Processing
+{
+    /// <summary>
+    /// Represents an implementation of <see cref="IDetectEmbeddingLoop" />.
+    /// </summary>
+    public class EmbeddingLoopDetector : IDetectEmdbeddingLoop
+    {
+        readonly ICompareStates _comparer = new CompareProjectionStates();
```

Comment:
This should be taken in as a dependency in the constructor

Reply:
I changed it to take it as a dependency and added mocks for it in the tests.

While doing this, I also realized that the `ParallelEnumerable.Any()` does not short circuit on exceptions/`true` results, but this is a question about optimizing and if even using parallelization gives us anything here. Time will tell.